analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/DeconstructionAssignment.cs

```diff
@@ -0,0 +1,89 @@
+ */
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors;
+
+internal sealed class DeconstructionAssignment : SimpleProcessor<IDeconstructionAssignmentOperationWrapper>
+{
+    protected override IDeconstructionAssignmentOperationWrapper Convert(IOperation operation) =>
+        IDeconstructionAssignmentOperationWrapper.FromOperation(operation);
+
+    protected override ProgramState Process(SymbolicContext context, IDeconstructionAssignmentOperationWrapper assignment)
+    {
+        var operationValues = CollectOperationValues(context.State, assignment.Target, assignment.Value);
+        var newState = context.State;
+        foreach (var (tupleMember, value) in operationValues)
+        {
+            newState = newState.SetOperationAndSymbolValue(tupleMember, value);
+        }
+        return newState;
+    }
+
+    private static IEnumerable<OperationValue> CollectOperationValues(ProgramState state, IOperation target, IOperation value)
+    {
+        var operationValues = new List<OperationValue>();
+        var leftTupleElements = TupleElements(target, state);
+        // If the right side is a tuple, then every symbol/constraint is copied to the left side.
+        if (Unwrap(value, state).AsTuple() is { } rightSideTuple)
+        {
+            var rightTupleElements = TupleElements(rightSideTuple.WrappedOperation, state);
+            for (var i = 0; i < leftTupleElements.Length; i++)
+            {
+                var leftTupleMember = leftTupleElements[i];
+                var rightTupleMember = rightTupleElements[i];
+                if (leftTupleMember.Kind == OperationKindEx.Discard)
+                {
+                    continue;
+                }
+                if (leftTupleMember.AsTuple() is { } nestedTuple)
+                {
+                    operationValues.AddRange(CollectOperationValues(state, nestedTuple.WrappedOperation, rightTupleMember));
+                }
+                else
+                {
+                    operationValues.Add(new(leftTupleMember, state[rightTupleMember]));
+                }
+            }
+        }
+        // If the right side is not a tuple, then every member of the left side tuple is set to empty.
+        else
+        {
+            operationValues.AddRange(leftTupleElements
+                .Where(x => x.Kind != OperationKindEx.Discard)
+                .Select(x => new OperationValue(x, SymbolicValue.Empty)));
+        }
+        return operationValues;
+    }
+
+    private static IOperation[] TupleElements(IOperation operation, ProgramState state) =>
+        Unwrap(operation, state)
+            .ToTuple()
+            .Elements
+            .Select(x => Unwrap(x, state))
+            .ToArray();
+
+    private static IOperation Unwrap(IOperation operation, ProgramState state)
+    {
+        var unwrapped = state.ResolveCaptureAndUnwrapConversion(operation);
+        return unwrapped.AsDeclarationExpression()?.Expression ?? unwrapped;
```

Comment:
When and why is this extra step needed?

Reply:
The `Target` of the `DeconstructionAssignment` can wrapped inside a variable declaration. [See this example.](https://edotor.net/?engine=dot#%2F%2A%0D%0Apublic%20void%20NestedDeconstructionAssignment%28%29%0D%0A%7B%0D%0A%20%20%20%20var%20%28a%2C%20%28b%2C%20_%29%29%20%3D%20%28true%2C%20%28true%2C%20true%29%29%3B%0D%0A%20%20%20%20if%20%28a%29%20%7B%20%7D%20%20%2F%2F%20Noncompliant%0D%0A%20%20%20%20if%20%28b%29%20%7B%20%7D%20%20%2F%2F%20Noncompliant%0D%0A%7D%0D%0A%2A%2F%0D%0A%0D%0Adigraph%20%22RoslynCFG%22%20%7B%0D%0Asubgraph%20%22cluster_1%22%20%7B%0Alabel%20%3D%20%22LocalLifetime%20region%22%0D%0Acfg0_block1%20%5Bshape%3Drecord%20label%3D%22%7BBLOCK%20%231%7D%22%5D%0D%0A%7D%0D%0Acfg0_block0%20%5Bshape%3Drecord%20label%3D%22%7BENTRY%20%230%7D%22%5D%0D%0Acfg0_block2%20%5Bshape%3Drecord%20label%3D%22%7BEXIT%20%232%7D%22%5D%0D%0Asubgraph%20%22cluster_3%22%20%7B%0Alabel%20%3D%20%22RoslynCFG.NestedDeconstructionAssignment%22%0D%0Asubgraph%20%22cluster_4%22%20%7B%0Alabel%20%3D%20%22LocalLifetime%20region%2C%20Locals%3A%20a%2C%20b%22%0D%0Acfg2_block1%20%5Bshape%3Drecord%20label%3D%22%7BBLOCK%20%231%7C0%23%3A%20ExpressionStatementOperation%3A%20var%20%28a%2C%20%28b%2C%20_%29%29%20%3D%20%28true%2C%20%28true%2C%20true%29%29%3B%7C1%23%3A%200%23.Operation%3A%20DeconstructionAssignmentOperation%3A%20var%20%28a%2C%20%28b%2C%20_%29%29%20%3D%20%28true%2C%20%28true%2C%20true%29%29%7C2%23%3A%201%23.Target%3A%20DeclarationExpressionOperation%3A%20var%20%28a%2C%20%28b%2C%20_%29%29%7C3%23%3A%202%23.Expression%3A%20TupleOperation%3A%20%28a%2C%20%28b%2C%20_%29%29%7C4%23%3A%20LocalReferenceOperation%3A%20a%7C4%23%3A%20TupleOperation%3A%20%28b%2C%20_%29%7C5%23%3A%20LocalReferenceOperation%3A%20b%7C5%23%3A%20DiscardOperation%3A%20_%7C2%23%3A%201%23.Value%3A%20TupleOperation%3A%20%28true%2C%20%28true%2C%20true%29%29%7C3%23%3A%20LiteralOperation%3A%20true%7C3%23%3A%20TupleOperation%3A%20%28true%2C%20true%29%7C4%23%3A%20LiteralOperation%3A%20true%7C4%23%3A%20LiteralOperation%3A%20true%7C%23%23%23%23%23%23%23%23%23%23%7C%23%23%20BranchValue%20%23%23%7C0%23%3A%20LocalReferenceOperation%3A%20a%7C%23%23%23%23%23%23%23%23%23%23%7D%22%5D%0D%0Acfg2_block2%20%5Bshape%3Drecord%20label%3D%22%7BBLOCK%20%232%7C%23%23%20BranchValue%20%23%23%7C0%23%3A%20LocalReferenceOperation%3A%20b%7C%23%23%23%23%23%23%23%23%23%23%7D%22%5D%0D%0A%7D%0D%0Acfg2_block0%20%5Bshape%3Drecord%20label%3D%22%7BENTRY%20%230%7D%22%5D%0D%0Acfg2_block3%20%5Bshape%3Drecord%20label%3D%22%7BEXIT%20%233%7D%22%5D%0D%0A%7D%0D%0Acfg0_block0%20-%3E%20cfg0_block1%0D%0Acfg0_block1%20-%3E%20cfg0_block2%0D%0Acfg2_block0%20-%3E%20cfg2_block1%0D%0Acfg2_block1%20-%3E%20cfg2_block2%20%5Blabel%3D%22WhenFalse%22%5D%0D%0Acfg2_block1%20-%3E%20cfg2_block2%20%5Blabel%3D%22Else%22%5D%0D%0Acfg2_block2%20-%3E%20cfg2_block3%20%5Blabel%3D%22WhenFalse%22%5D%0D%0Acfg2_block2%20-%3E%20cfg2_block3%20%5Blabel%3D%22Else%22%5D%0D%0A%7D%0D%0A)